### DISCUSSION TITLE
Unexpected behaviour of packageRules.matchFileNames

### DISCUSSION THREAD

**Discussion by jirkamelichar** [renovatebot/renovate#30636] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, Renovate v38.20.1

### Please tell us more about your question or problem

I have the following configuration in place:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:recommended"
  ],
  "enabledManagers": ["npm"],
  "packageRules": [
    {
      "matchFileNames": ["/lib-a/**", "/lib-b/**"],
      "matchManagers": ["npm"],
      "matchUpdateTypes": ["minor", "patch"]
    }
  ]
}
```

What I expect is that Renovate will only update `package.json` dependencies within the 2 folders (`lib-a` and `lib-b`). This is a monorepo, which contains a number of other folders with `package.json` files.

I've noticed that Im getting an MR from Renovate which contains an update in `lib-b/package.json` (as expected), but also, in that same MR it is updating `another-lib/package.json` and `yet-another-lib/package.json` - that is breaking the `matchFileNames` rule. Note that the MR is dealing with a single dependency across all the affected `package.json`'s

Is this the expected behaviour or have I got something wrong in my configuration?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
You've got something wrong. A package rule with only "match" elements does nothing. If your goal is to disable certain dependencies or updates then you need at least one rule with enabled=false in order to disable things. The default behavior of renovate is to update everything. Look at any package rules in docs or this repo for examples
